Challenge1/src/agent.spec.ts

```diff
@@ -1,74 +1,108 @@
-// import {
-//   FindingType,
-//   FindingSeverity,
-//   Finding,
-//   HandleTransaction,
-//   createTransactionEvent,
-//   ethers,
-// } from "forta-agent";
-// import agent, {
-//   ERC20_TRANSFER_EVENT,
-//   TETHER_ADDRESS,
-//   TETHER_DECIMALS,
-// } from "./agent";
-
-// describe("high tether transfer agent", () => {
-//   let handleTransaction: HandleTransaction;
-//   const mockTxEvent = createTransactionEvent({} as any);
-
-//   beforeAll(() => {
-//     handleTransaction = agent.handleTransaction;
-//   });
-
-//   describe("handleTransaction", () => {
-//     it("returns empty findings if there are no Tether transfers", async () => {
-//       mockTxEvent.filterLog = jest.fn().mockReturnValue([]);
-
-//       const findings = await handleTransaction(mockTxEvent);
-
-//       expect(findings).toStrictEqual([]);
-//       expect(mockTxEvent.filterLog).toHaveBeenCalledTimes(1);
-//       expect(mockTxEvent.filterLog).toHaveBeenCalledWith(
-//         ERC20_TRANSFER_EVENT,
-//         TETHER_ADDRESS
-//       );
-//     });
-
-//     it("returns a finding if there is a Tether transfer over 10,000", async () => {
-//       const mockTetherTransferEvent = {
-//         args: {
-//           from: "0xabc",
-//           to: "0xdef",
-//           value: ethers.BigNumber.from("20000000000"), //20k with 6 decimals
-//         },
-//       };
-//       mockTxEvent.filterLog = jest
-//         .fn()
-//         .mockReturnValue([mockTetherTransferEvent]);
-
-//       const findings = await handleTransaction(mockTxEvent);
-
-//       const normalizedValue = mockTetherTransferEvent.args.value.div(
-//         10 ** TETHER_DECIMALS
-//       );
-//       expect(findings).toStrictEqual([
-//         Finding.fromObject({
-//           name: "High Tether Transfer",
-//           description: `High amount of USDT transferred: ${normalizedValue}`,
-//           alertId: "FORTA-1",
-//           severity: FindingSeverity.Low,
-//           type: FindingType.Info,
-//           metadata: {
-//             to: mockTetherTransferEvent.args.to,
-//             from: mockTetherTransferEvent.args.from,
-//           },
-//         }),
-//       ]);
-//       expect(mockTxEvent.filterLog).toHaveBeenCalledTimes(1);
-//       expect(mockTxEvent.filterLog).toHaveBeenCalledWith(
-//         ERC20_TRANSFER_EVENT,
-//         TETHER_ADDRESS
-//       );
-//     });
-//   });
-// });
+import { Finding, HandleTransaction, FindingSeverity, FindingType, TransactionEvent } from "forta-agent";
+import agent from "./agent";
+
+let handleTransaction: HandleTransaction;
+const mockNetherMindAddress = "0x88dC3a2284FA62e0027d6D6B1fCfDd2141a143b8";
+const mockFortaRegistryAddress = "0x61447385B019187daa48e91c55c02AF1F1f3F863";
```

Comment:
In the test file, we usually mock all of the values we can. Also, take a look at our `forta-agent-tools` (aka GAM, for general agents module) library for some testing tools (these were developed in-house by an early Forta team at Nethermind!). Library is [here](https://github.com/NethermindEth/general-agents-module). The one that would be relevant here would be `createAddress()`.

Reply:
^This comment is unresolved.

Reply:
The naming was misleading. These are actually the real addresses, so we can test for cases where those constraints are met - e.g. tx from Nethermind address but **not** to Forta registry address - use the mockNetherMindAddress for that. I renamed the variables now to netherMindAddress and fortaRegistryAddress

Reply:
I am now using gam createAddress() wherever we want a random address

Reply:
@deca12x, as you correctly wrap the `handleTransaction` into a `provideHandleTransaction` that takes the two addresses as parameters, you can call `provideHandleTransaction` with the mock addresses in your test and they will work as expected.

Reply:
I'm confused, I thought in the test we are:
- defining a tx (from address, to address, function call)
- ...and running the bot on that tx.

So essentially, after defining the conditions of the tx, we compare it to the conditions that the bot filters by
So we still need to define the two real addresses in the test, in order to check that the bot will recognise those as the correct ones. Is my understanding inaccurate?